content/docs/07-reference/01-ai-sdk-core/31-ui-message.mdx

```diff
@@ -36,9 +36,9 @@ const dataPartSchema = z.object({
 
 type MyDataPart = z.infer<typeof dataPartSchema>;
 
-const tools: ToolSet = {
+const tools = {
   someTool: tool({}),
-};
+} satisfies ToolSet;
```

Comment:
Without this, the tool won't be inferred.

Reply:
I think the key was removing `: ToolSet` which erases the generics. Does the satisfy impact the typing?

Reply:
(and docs should ideally be minimal)